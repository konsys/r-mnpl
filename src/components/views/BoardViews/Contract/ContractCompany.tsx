import { IField } from "../../../../types/types";
import React from "react";

export const ContractCompany = ({ field }: { field: IField }) => {
  const price = `${field.price?.startPrice}k` || "0k";
  return (
    <>
      {field.fieldId && (
        <div className="_one _field _deletable">
          <div
            className="_image"
            style={{
              backgroundImage: `url("${field.imgSrc}")`,
            }}
          ></div>
          <div className="_info">
            <div className="_title">{field.name}</div>
            <div className="_subtitle">{price}</div>
          </div>
        </div>
      )}
    </>
  );
};
