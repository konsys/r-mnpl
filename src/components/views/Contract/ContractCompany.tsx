import { IField } from "../../../types/types";
import React from "react";

export const ContractCompany = ({ field }: { field: IField }) => {
  return (
    <>
      {field.fieldId && (
        <div className="_one _field _deletable">
          <div
            className="_image"
            style={{
              backgroundImage:
                'url("https://m1.dogecdn.wtf/fields/brands/2_clothing/lacoste.svg")',
            }}
          ></div>
          <div className="_info">
            <div className="_title">{field.name}</div>
            <div className="_subtitle">{field.price?.startPrice}k</div>
          </div>
        </div>
      )}
    </>
  );
};
