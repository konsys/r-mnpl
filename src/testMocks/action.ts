import { ICurrentAction } from "stores/Board/ActionStore";
import {
  BoardAction,
  IncomeMessageType,
  OutcomeMessageType,
} from "types/types";

export const testDoNothingAction: ICurrentAction = {
  actionId: "testActionId",
  event: {
    action: {
      _id: "testID",
      isModal: false,
      userId: 1,
      type: IncomeMessageType.DO_NOTHING,
    },
  },
};

export const testRollDicesModal: BoardAction = {
  _id: "testBoardAction",
  type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,

  userId: 1,
  title: "testTitleRollDices",
  text: "testTextRollDices",
  isModal: true,
};
export const testRollDicesClicked: BoardAction = {
  _id: "testBoardAction",
  type: OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED,
  userId: 1,
  isModal: false,
};
