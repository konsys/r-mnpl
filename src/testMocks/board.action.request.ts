import { IBoardActionRequest, OutcomeMessageType } from "types/types";

export const testBoardActionRequest:IBoardActionRequest = {
    gameId:'testGameId',
    action:OutcomeMessageType.OUTCOME_ROLL_DICES_CLICKED
}