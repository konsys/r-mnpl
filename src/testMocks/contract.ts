import { IContract } from "types/types";

export const testContract: IContract = {
  fieldFromPrice: 1000,
  fieldIdsFrom: [1, 2],
  fieldIdsTo: [3],
  fieldToPrice: 1000,
  fromUserId: 1,
  moneyFrom: 1000,
  moneyTo: 1000,
  toUserId: 2,
};
