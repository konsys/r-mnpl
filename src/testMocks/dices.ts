import { IDices, IncomeMessageType } from "types/types";

export const testDices: IDices = {
    _id: "testId",
    isModal: true,
    type: IncomeMessageType.INCOME_ROLL_DICES_MODAL,
    dices: [1, 2, 3],
    dicesSum: 6,
    userId: 1,
  };
  