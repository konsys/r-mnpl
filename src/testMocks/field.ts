import { FieldType, IField } from "types/types";

export const testField: IField = {
  fieldId: 1,
  fieldPosition: 1,
  name: "testName",
  type: FieldType.COMPANY,
  price: {
    branchPrice: 1000,
    buyoutPrice: 900,
    pledgePrice: 500,
    startPrice: 2000,
  },
  rent: {
    baseRent: 200,
    monopolyRent: 250,
    oneStar: 300,
    twoStar: 1000,
    freeStar: 1500,
    fourStar: 2000,
    bigStar: 3000,
  },
};
