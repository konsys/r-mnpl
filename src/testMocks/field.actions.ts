import { testField, testMortgagedField, testOwnedField } from "./field";

import { IFieldModal } from "components/views/BoardViews/Field/FieldActions/FieldActions";

export const testFieldActions: IFieldModal = {
  ...testField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testOwnedFieldActions: IFieldModal = {
  ...testOwnedField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testStaredFieldActions: IFieldModal = {
  ...testOwnedField,
  isActive: true,

  position: { top: 20, left: 20 },
};

export const testMortgagedFieldActions: IFieldModal = {
  ...testMortgagedField,
  isActive: true,

  position: { top: 20, left: 20 },
};
