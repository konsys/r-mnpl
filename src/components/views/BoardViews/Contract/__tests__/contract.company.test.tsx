import { ContractCompany } from "../ContractCompany";
import React from "react";
import { shallow } from "enzyme";
import { testField } from "testMocks/field";

describe("Test contract company", () => {
  it("should render correctly", () => {
    shallow(<ContractCompany field={testField} />);
  });
  it("should snapshot be the same", () => {
    const tree = shallow(<ContractCompany field={testField} />);
    expect(tree).toMatchSnapshot();
  });

  it("should render name", () => {
    const tree = shallow(<ContractCompany field={testField} />);
    expect(tree.find("._title").contains(testField.name)).toBeTruthy();
  });

  it("should render name", () => {
    const tree = shallow(<ContractCompany field={testField} />);
    const price = testField && testField.price && testField.price.startPrice;
    expect(tree.find("._subtitle").contains(`${price}k`)).toBeTruthy();
  });
});
