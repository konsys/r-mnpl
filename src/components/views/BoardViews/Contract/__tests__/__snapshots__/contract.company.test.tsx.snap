// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test contract company should snapshot be the same 1`] = `ShallowWrapper {}`;
